pubsub_logging/utils.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for the Python logging handlers."""
+
+import base64
+import os
+import sys
+import threading
+import time
+
+from googleapiclient import discovery
+from googleapiclient import errors
+
+import httplib2
+
+from oauth2client.client import GoogleCredentials
+
+from pubsub_logging.errors import RecoverableError
+
+
+PUBSUB_SCOPES = ["https://www.googleapis.com/auth/pubsub"]
+
+clients = threading.local()
+
+
+def compat_urlsafe_b64encode(v, force_py3_mode=False):
+    """A urlsafe ba64encode which is compatible with Python 2 and 3.
+
+    Args:
+      v: A string to encode.
+      force_py3_mode: If True, it will use the code for Python3.
+    Returns:
+      The encoded string.
+    """
+    if sys.version_info[0] == 3 or force_py3_mode:
+        return base64.urlsafe_b64encode(v.encode('UTF-8')).decode('ascii')
+    else:
+        return base64.urlsafe_b64encode(v)
+
+
+def get_pubsub_client(http=None):  # pragma: NO COVER
+    """Return a thread local Pub/Sub client.
+
+    Args:
+      http: httplib2.Http instance. Defaults to None.
+    Returns:
+      Cloud Pub/Sub client.
+    """
+    if not hasattr(clients, 'client'):
+        credentials = GoogleCredentials.get_application_default()
+        if credentials.create_scoped_required():
+            credentials = credentials.create_scoped(PUBSUB_SCOPES)
```

Comment:
Makes sense, however I think that's a feature request for the auth library.
The situation here is way better than before.